app/components/primer/select_menu_component.rb

```diff
@@ -0,0 +1,168 @@
+# frozen_string_literal: true
+
+module Primer
+  class SelectMenuComponent < Primer::Component
+    include ViewComponent::Slotable
+
+    DEFAULT_ALIGN_RIGHT = false
+
+    with_slot :header, class_name: "Header"
+    with_slot :modal, class_name: "Modal"
+    with_slot :filter, class_name: "Filter"
+    with_slot :footer, class_name: "Footer"
+    with_slot :close_button, class_name: "CloseButton"
+
+    def initialize(align_right: DEFAULT_ALIGN_RIGHT, **kwargs)
+      @align_right = fetch_or_fallback([true, false], align_right, DEFAULT_ALIGN_RIGHT)
+      @kwargs = kwargs
+      @kwargs[:tag] = :div
+    end
+
+    def render?
+      modal.present?
+    end
+
+    class Modal < Primer::Slot
+      BORDER_CLASSES = {
+        all: nil,
+        omit_top: "border-top-0",
+        none: "SelectMenu-list--borderless",
+      }.freeze
+      DEFAULT_BORDER_CLASS = :all
+
+      attr_reader :message
+
+      def initialize(border: DEFAULT_BORDER_CLASS, message: nil, **kwargs)
+        @border = fetch_or_fallback(BORDER_CLASSES.keys, border, DEFAULT_BORDER_CLASS)
+        @message = message
+        @kwargs = kwargs
+        @kwargs[:tag] = :div
+        @kwargs[:classes] = class_names(
+          "SelectMenu-modal",
+          kwargs[:classes],
+        )
+      end
+
+      def outer_component
+        Primer::BaseComponent.new(**@kwargs)
+      end
+
+      def inner_component
+        Primer::BaseComponent.new(
+          tag: :div,
+          role: "menu",
+          classes: class_names(
+            "SelectMenu-list",
+            @kwargs[:list_classes],
+            BORDER_CLASSES[@border],
+          )
+        )
+      end
+
+      def message_component
+        Primer::BaseComponent.new(
+          tag: :div,
+          classes: class_names(
+            "SelectMenu-message",
+            @kwargs[:message_classes],
+          )
+        )
+      end
+    end
+
+    class Header < Primer::Slot
+      def initialize(**kwargs)
+        @kwargs = kwargs
+        @kwargs[:tag] ||= :header
+        @kwargs[:classes] = class_names(
+          "SelectMenu-header",
+          kwargs[:classes]
+        )
+      end
+
+      def outer_component
+        Primer::BaseComponent.new(**@kwargs)
+      end
+
+      def inner_component
```

Comment:
not sure how I feel about `inner/outer` components, it looks too abstract and makes it harder to read imo

I think that making it more specific will help others => `wrapper_component` and `title_component`

Reply:
Ooh gotcha, I can do that!